### PR TITLE
[GCI] Standardised Edge Detect module code comments

### LINKED ISSUE #1341: Standardise Edge Detect module code comments
Note: **We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more.**

This is a sub-issue of #1046

The comments in the following files are inconsistent:
1. [Module.js](https://github.com/publiclab/image-sequencer/blob/src/modules/EdgeDetect/Module.js)
2. [EdgeUtils.js](https://github.com/publiclab/image-sequencer/blob/main/src/modules/EdgeDetect/EdgeUtils.js)

Rules:
1. Methods must be described using [JSDoc comments](https://devdocs.io/jsdoc/)
2. Misc code comments should be inline unless it is a long sentence.
3. No use of continuous tenses, no pronouns.
4. No redundant comments.
5. Each comment should start with an uppercase letter and end with a full stop.

### PR DESCRIPTION
**Pull request by HarshKhandeparkar**
Fixes #1341  (<=== Replace `0000` with the Issue Number)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 1]
```diff
@@ -32,28 +34,47 @@ module.exports = function(pixels, highThresholdRatio, lowThresholdRatio, useHyst
       angles.slice(-1)[0].push(result.angle);
     }
   }
-  nonMaxSupress(pixels, grads, angles);
-  doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels);
-  if(useHysteresis.toLowerCase() == 'true') hysteresis(strongEdgePixels, weakEdgePixels);
+  nonMaxSupress(pixels, grads, angles); // Non Maximum Suppression: Filter fine edges.
+  doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels); // Double Threshold: Categorizes edges into strong and weak edges based on two thresholds.
+  if(useHysteresis.toLowerCase() == 'true') hysteresis(strongEdgePixels, weakEdgePixels); // Optional Hysteresis to minimize edges generated due to noise. This process is quite slow.
```
Comment (SidharthBansal) [inline src/modules/EdgeDetect/EdgeUtils.js]: This can't be used

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 2]
```diff
@@ -76,14 +97,20 @@ function sobelFilter(pixels, x, y) {
     }
   }
 
-  const grad = Math.sqrt(Math.pow(gradX, 2) + Math.pow(gradY, 2)),
+  const grad = Math.sqrt(Math.pow(gradX, 2) + Math.pow(gradY, 2)), // gradient(magnitude)
```
Comment (SidharthBansal) [inline src/modules/EdgeDetect/EdgeUtils.js]: Don't write self-descriptive comments. Please remove

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 3]
```diff
@@ -147,17 +182,24 @@ function nonMaxSupress(pixels, grads, angles) {
...
+// Finds the max value in a 2d array like grads.
 var findMaxInMatrix = arr => Math.max(...arr.map(el => el.map(val => val ? val : 0)).map(el => Math.max(...el)));
 
-// Applies the double threshold to the image
+// Applies the double threshold to the image.
 function doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels) {
 
-  const highThreshold = findMaxInMatrix(grads) * highThresholdRatio,
-    lowThreshold = highThreshold * lowThresholdRatio;
+  const highThreshold = findMaxInMatrix(grads) * highThresholdRatio, // High Threshold relative to the strongest edge.
```
Comment (SidharthBansal) [inline src/modules/EdgeDetect/EdgeUtils.js]: I think you forget to remove full stop from phrases in this pr

Comment (HarshKhandeparkar) [reply src/modules/EdgeDetect/EdgeUtils.js]: Yes, I had forgotten to do that but I have pushed a fix now.

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 4]
```diff
@@ -198,8 +198,8 @@ var findMaxInMatrix = arr => Math.max(...arr.map(el => el.map(val => val ? val :
 // Applies the double threshold to the image.
 function doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels) {
 
-  const highThreshold = findMaxInMatrix(grads) * highThresholdRatio, // High Threshold relative to the strongest edge.
-    lowThreshold = highThreshold * lowThresholdRatio; // Low threshold relative to high threshold.
+  const highThreshold = findMaxInMatrix(grads) * highThresholdRatio, // High Threshold relative to the strongest edge
```
Comment (SidharthBansal) [inline src/modules/EdgeDetect/EdgeUtils.js]: T->t

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 5]
```diff
@@ -21,7 +23,7 @@ module.exports = function(pixels, highThresholdRatio, lowThresholdRatio, useHyst
     grads.push([]);
     angles.push([]);
     for (var y = 0; y < pixels.shape[1]; y++) {
-      var result = sobelFilter(
+      var result = sobelFilter( // Convolves the sobel filter on every pixel.
```
Comment (SidharthBansal) [inline src/modules/EdgeDetect/EdgeUtils.js]: Please remove full stop from phrases

Comment (HarshKhandeparkar) [reply src/modules/EdgeDetect/EdgeUtils.js]: Why is that considered as a phrase?

File: src/modules/EdgeDetect/EdgeUtils.js (hunk 4 above)
Comment (HarshKhandeparkar) [reply src/modules/EdgeDetect/EdgeUtils.js]: `High Threshold` is a name, a proper name for the High Threshold. It is not a generic high threshold, it is the Canny Edge Detection High Threshold.

File: src/modules/EdgeDetect/EdgeUtils.js (hunk 5 above)
Comment (SidharthBansal) [reply src/modules/EdgeDetect/EdgeUtils.js]: I supposed that a sentence consists of a subject and predicate. Here only predicate is present, hence I thought that it is a phrase.

File: src/modules/EdgeDetect/EdgeUtils.js (hunk 4 above)
Comment (SidharthBansal) [reply src/modules/EdgeDetect/EdgeUtils.js]: Then it is good !

**Comment by HarshKhandeparkar** [comment]
Hmm. I am not very good at English. I asked one the dev of a library who
has experience in development field and also happens to be very good ar
literature said that any group of words can be considered a phrase and the
full stop makes it a sentence. That is an infinite loop so I think making
it a phrase would be the best option. He said that this one is both a
sentence and a phrase which blew my mind.

**Comment by SidharthBansal** [comment]
It is a debatable thing. Most developers don't use a full stop in getter and setter things.  As they are in a hurry. So, I asked to remove unnecessary full stops. This will others to not write full stops in every single comment.

File: src/modules/EdgeDetect/EdgeUtils.js (hunk 5 above)
Comment (HarshKhandeparkar) [reply src/modules/EdgeDetect/EdgeUtils.js]:
```suggestion
      var result = sobelFilter( // Convolves the sobel filter on every pixel
```

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 6]
```diff
@@ -32,28 +34,47 @@ module.exports = function(pixels, highThresholdRatio, lowThresholdRatio, useHyst
       angles.slice(-1)[0].push(result.angle);
     }
   }
-  nonMaxSupress(pixels, grads, angles);
-  doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels);
-  if(useHysteresis.toLowerCase() == 'true') hysteresis(strongEdgePixels, weakEdgePixels);
+  nonMaxSupress(pixels, grads, angles); // Non Maximum Suppression: Filter fine edges.
+  doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels); // Double Threshold: Categorizes edges into strong and weak edges based on two thresholds.
+  if(useHysteresis.toLowerCase() == 'true') hysteresis(strongEdgePixels, weakEdgePixels); // Optional Hysteresis (very slow) to minimize edges generated due to noise
```
Comment (HarshKhandeparkar) [inline src/modules/EdgeDetect/EdgeUtils.js]:
```suggestion
  if(useHysteresis.toLowerCase() == 'true') hysteresis(strongEdgePixels, weakEdgePixels); // Optional Hysteresis (very slow) to minimize edges generated due to noise.
```

File: src/modules/EdgeDetect/EdgeUtils.js [hunk 7]
```diff
@@ -65,8 +86,8 @@ function sobelFilter(pixels, x, y) {
       let xn = x + a - 1,
         yn = y + b - 1;
 
-      if (isOutOfBounds(pixels, xn, yn)) {
-        gradX += pixels.get(xn + 1, yn + 1, 0) * kernelx[a][b];
+      if (isOutOfBounds(pixels, xn, yn)) { // Fallback for coordinates which lie outside the image.
+        gradX += pixels.get(xn + 1, yn + 1, 0) * kernelx[a][b]; // Fallback to nearest pixel.
```
Comment (HarshKhandeparkar) [inline src/modules/EdgeDetect/EdgeUtils.js]:
```suggestion
        gradX += pixels.get(xn + 1, yn + 1, 0) * kernelx[a][b]; // Fallback to nearest pixel
```

**Comment by jywarren** [comment]
Looking at this one now!

**Pull request merged**